# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/tests/lib/tests.py
# Commit: 9d31a829691c
# Full Hash: 9d31a829691ccf798375446e7ef04109e2662175
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2023-06-07 03:30:33
# Regressor Bug: 1835876
# File Overlap Count: 1
# Description:
#   Bug 1835876 part 2 - Disable code write protection in content processes. r=nbp
#   
#   We've worked with the security and performance teams to re-evaluate the W^X policy
#   we have in place to mark JIT code memory pages as either writable or executable
#   (but not both).
# ==============================================================================

diff -r 1ce5ad17da6b -r 9d31a829691c js/src/tests/lib/tests.py
--- a/js/src/tests/lib/tests.py	Tue Jun 06 12:16:46 2023 +0000
+++ b/js/src/tests/lib/tests.py	Tue Jun 06 12:16:46 2023 +0000
@@ -25,7 +25,7 @@
             "--no-sse3",
             "--no-threads",
         ],
-        ["--baseline-eager"],
+        ["--baseline-eager", "--write-protect-code=off"],
         ["--no-blinterp", "--no-baseline", "--no-ion", "--more-compartments"],
         ["--blinterp-eager"],
     ],
@@ -38,12 +38,12 @@
             "--ion-offthread-compile=off",  # implies --baseline-eager
             "--more-compartments",
         ],
-        ["--baseline-eager"],
+        ["--baseline-eager", "--write-protect-code=off"],
         ["--no-blinterp", "--no-baseline", "--no-ion", "--more-compartments"],
     ],
     # used by jit_test.py
     "ion": [
-        ["--baseline-eager"],
+        ["--baseline-eager", "--write-protect-code=off"],
         ["--ion-eager", "--ion-offthread-compile=off", "--more-compartments"],
     ],
     # Run reduced variants on debug builds, since they take longer time.
@@ -54,7 +54,7 @@
             "--ion-offthread-compile=off",  # implies --baseline-eager
             "--more-compartments",
         ],
-        ["--baseline-eager"],
+        ["--baseline-eager", "--write-protect-code=off"],
     ],
     # Cover cases useful for tsan. Note that we test --ion-eager without
     # --ion-offthread-compile=off here, because it helps catch races.